mkosi/__init__.py

```diff
@@ -5650,13 +5688,27 @@ def build_auxiliary_output_path(args: argparse.Namespace, suffix: str, can_compr
     return output.with_name(f"{output.name}{suffix}{compression or ''}")
 
 
-def check_valid_script(path: Path) -> None:
+DISABLED = Path('DISABLED')  # a placeholder value to suppress autodetection
+
+def script_path(value: Optional[str]) -> Optional[Path]:
+    if value is None:
+        return None
+    if value == '':
+        return DISABLED
+    return Path(value)
+
+
+def normalize_script(path: Optional[Path]) -> Optional[Path]:
+    if not path or path is DISABLED:
```

Comment:
```suggestion
    if not path or path == DISABLED:
```
Even though the string might get interred and this could work in that case (don't really know), I generally get
```
>>> a = Path('DISABLED')
>>> b = Path('DISABLED')
>>> a is b
False
```

Reply:
Internment of strings is not relevant here: we are comparing Path objects. And I used `is` on purpose: `DISABLED` is created once in the module, and we always use this particular object in the code. If we were to recreate the object in a new place, `is` would not work, and `==` would be required. But I don't think we want to allow this: the original object should always be used. In fact, if I specify `--finalize-script=DISABLED`, this will be handled like a normal path, which I think is good.

(Let's say that I want to reuse this logic in `backend.py`:
```
from mkosi import DISABLED
x: Path = DISABLED
```
but not
```
x: Path = Path('DISABLED')
```)

(Overall, I'd prefer to create a real singleton object for this. But Python doesn't really make this easy. A different option would be to create a non-Path object, e.g. `DISABLED = object()` and use that as the placeholder. But this would be a typing violation, and I don't think it's worth it to adjust the code to allow a different type. In practice, the thing with Path('DISABLED') should just work without any issues except theoretical concerns.)

Reply:
Ah, I see, you're right. No problem then.